app/src/data/schemas/searchSchema.ts

```diff
@@ -0,0 +1,64 @@
+const searchSchema = {
+  response: {
+    // Keyword included in search
+    keyword: "string",
+    // Pagination fields
+    numResults: "integer",
+    page: "integer",
+    perPage: "integer",
+    // Sort fields
+    sort: "string ie: title DESC",
+    // Filter fields
+    filterField: "string",
+    // Results
+    results: [
+      {
```

Comment:
after reading the requirements for the modal in the brd, I think we can include the list of filter categories and the first 10 filters per category in this endpoint. each category would have a field for the total count of filters so the app knows when to display the "View all" button that loads the modal.

Reply:
I'm not sure about this and I want you to talk to the backend team about this. This is verging on writing and getting data from the API _just_ for the frontend. What if in the future, the modal approach does not work and the frontend needs to display 5 or 20 filters? I'm not disagreeing with this approach but just want acknowledgment from everyone that this is the right way to do this.

What I find helpful from APIs is a "self" link to more resources. So, besides the total count of filters property, there can be a property for "self" that is another endpoint to get the rest of the filters. Otherwise, how would you know where to fetch the filters needed? Just an idea.

Reply:
this would be on par with what the research catalog does https://www.nypl.org/research/research-catalog/api/search?q=cats